### DISCUSSION TITLE
How to populate form on initial page load with server data?

### DISCUSSION THREAD

**Discussion by shafkathullah** [houseform/houseform#94] [Q&A] [answered]
when I go to say page `example.com?id=g32hg` I need to load data corresponding to that specific `id` from server and populate the form. Whats the best way to do that?

**Comment by joaom00** [comment, marked as answer]
You can use the `initialValue` prop on `Field`.

**Reply by shafkathullah** [reply]
These values be only available after server request, so should i only rendering the form after these values are available?

**Reply by joaom00** [reply]
Yeah. For me, at least, it's a weird UX to have a blank form that gets filled in later. What if the request takes too long?

**Reply by shafkathullah** [reply]
makes sense 👍
